iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/hint/DocumentOrganizationBuilderTest.java

```diff
@@ -0,0 +1,184 @@
+package eu.dnetlib.iis.wf.affmatching.hint;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.api.java.function.PairFunction;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import eu.dnetlib.iis.wf.affmatching.model.DocumentOrganization;
+import eu.dnetlib.iis.wf.affmatching.model.DocumentProject;
+import eu.dnetlib.iis.wf.affmatching.model.ProjectOrganization;
+import scala.Tuple2;
+
+/**
+ * @author mhorst
+ *
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class DocumentOrganizationBuilderTest {
+
+    DocumentOrganizationBuilder builder = new DocumentOrganizationBuilder();
+
+    @Mock
+    private JavaRDD<DocumentProject> documentProjectRdd;
+
+    @Captor
+    private ArgumentCaptor<PairFunction<DocumentProject, String, DocumentProject>> pairDocumentProjectFunctionArg;
+
+    @Mock
+    private JavaPairRDD<String, DocumentProject> documentProjectPairedRdd;
+
+    @Captor
+    private ArgumentCaptor<Function<Tuple2<String, DocumentProject>, Boolean>> filterDocumentProjectFunctionArg;
+
+    @Mock
+    private JavaPairRDD<String, DocumentProject> documentProjectPairedFilteredRdd;
+
+    @Mock
+    private JavaRDD<ProjectOrganization> projectOrganizationRdd;
+
+    @Captor
+    private ArgumentCaptor<PairFunction<ProjectOrganization, String, ProjectOrganization>> pairProjectOrganizationFunctionArg;
+
+    @Mock
+    private JavaPairRDD<String, ProjectOrganization> projectOrganizationPairedRdd;
+
+    @Mock
+    private JavaPairRDD<String, Tuple2<DocumentProject, ProjectOrganization>> documentOrganizationJoinedRdd;
+
+    @Captor
+    private ArgumentCaptor<Function<Tuple2<String, Tuple2<DocumentProject, ProjectOrganization>>, DocumentOrganization>> mapDocumentOrganizationFunctionArg;
```

Comment:
Well, yes now it is a sort of a monster name :-)) I thought about smth simpler like createDocOrgMapFunction or something like this... ;-)